analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.cs

```diff
@@ -1,30 +1,122 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Text;
 
 namespace Tests.Diagnostics
 {
     public class S3267
     {
-        public List<string> SuggestWhere(ICollection<string> collection, Predicate<string> condition)
+        public bool WithOutArg(string p1, out string p2)
+        {
+            p2 = string.Empty;
+            return true;
+        }
+
+        public void ForEachWithBlockWithIfSuggestWhere(ICollection<string> collection, Predicate<string> condition)
         {
             var result = new List<string>();
 
             foreach (var element in collection) // Noncompliant {{Loops should be simplified with "LINQ" expressions}}
+            //                      ^^^^^^^^^^
```

Comment:
could you also add a test with

```
foreach (var element in collection)
{
  Foo(1);
  if (condition(element))
  {
    result.Add(element);
   }
}

```
and
could you also add a test with

```
foreach (var element in collection)
{
  result.Add(element);
  if (condition(element))
  {
  }
}
 
```

(you have with member access usage, but not with method call before/after the if)